File: templates/Uwp/_comp/_shared/Feat.IdentityOptionalLogin._VB/Param_ProjectName/Helpers/AuthenticationHelper.vb
Imports System
Imports System.Threading.Tasks
Imports Param_RootNamespace.Core.Helpers
Imports Windows.UI.Popups

Namespace Helpers
    Friend Module AuthenticationHelper
        Friend Async Function ShowLoginErrorAsync(loginResult As LoginResultType) As Task
            Select Case loginResult
                Case LoginResultType.NoNetworkAvailable
                    Await New MessageDialog("DialogNoNetworkAvailableContent".GetLocalized(), "DialogAuthenticationTitle".GetLocalized()).ShowAsync()
                Case LoginResultType.UnknownError
                    Await New MessageDialog("DialogStatusUnknownErrorContent".GetLocalized(), "DialogAuthenticationTitle".GetLocalized()).ShowAsync()
            End Select
        End Function
    End Module
End Namespace

File: templates/Wpf/Services/IdentityOptionalLogin/Helpers/AuthenticationHelper.cs
using System.Threading.Tasks;
using System.Windows;
using Param_RootNamespace.Core.Helpers;
using Param_RootNamespace.Strings;
using MahApps.Metro.Controls;
using MahApps.Metro.Controls.Dialogs;

namespace Param_RootNamespace.Helpers
{
    internal static class AuthenticationHelper
    {
        internal static async Task ShowLoginErrorAsync(LoginResultType loginResult)
        {
            var metroWindow = Application.Current.MainWindow as MetroWindow;
            switch (loginResult)
            {
                case LoginResultType.NoNetworkAvailable:
                    await metroWindow.ShowMessageAsync(Resources.DialogNoNetworkAvailableContent, Resources.DialogAuthenticationTitle);
                    break;
                case LoginResultType.UnknownError:
                    await metroWindow.ShowMessageAsync(Resources.DialogAuthenticationTitle, Resources.DialogStatusUnknownErrorContent);
                    break;
            }
        }
    }
}
